simapp/v2/sim_runner.go

```diff
@@ -0,0 +1,485 @@
+package simapp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"iter"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"slices"
+	"testing"
+	"time"
+
+	cmtproto "github.com/cometbft/cometbft/api/cometbft/types/v1"
+	cmttypes "github.com/cometbft/cometbft/types"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+
+	"cosmossdk.io/core/appmodule"
+	appmodulev2 "cosmossdk.io/core/appmodule/v2"
+	"cosmossdk.io/core/comet"
+	corecontext "cosmossdk.io/core/context"
+	"cosmossdk.io/core/server"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/depinject"
+	"cosmossdk.io/log"
+	"cosmossdk.io/runtime/v2"
+	serverv2 "cosmossdk.io/server/v2"
+	"cosmossdk.io/server/v2/appmanager"
+	cometbfttypes "cosmossdk.io/server/v2/cometbft/types"
+	storev2 "cosmossdk.io/store/v2"
+	consensustypes "cosmossdk.io/x/consensus/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simsx"
+	simsxv2 "github.com/cosmos/cosmos-sdk/simsx/v2"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation/client/cli"
+)
+
+type Tx = transaction.Tx
+type (
+	HasWeightedOperationsX              = simsx.HasWeightedOperationsX
+	HasWeightedOperationsXWithProposals = simsx.HasWeightedOperationsXWithProposals
+	HasProposalMsgsX                    = simsx.HasProposalMsgsX
+)
+
+const SimAppChainID = "simulation-app"
+
+// this list of seeds was imported from the original simulation runner: https://github.com/cosmos/tools/blob/v1.0.0/cmd/runsim/main.go#L32
+var defaultSeeds = []int64{
+	1, 2, 4, 7,
+	32, 123, 124, 582, 1893, 2989,
+	3012, 4728, 37827, 981928, 87821, 891823782,
+	989182, 89182391, 11, 22, 44, 77, 99, 2020,
+	3232, 123123, 124124, 582582, 18931893,
+	29892989, 30123012, 47284728, 7601778, 8090485,
+	977367484, 491163361, 424254581, 673398983,
+}
+
+const (
+	maxTimePerBlock   = 10_000 * time.Second
+	minTimePerBlock   = maxTimePerBlock / 2
+	timeRangePerBlock = maxTimePerBlock - minTimePerBlock
+)
+
+type (
+	AuthKeeper interface {
+		simsx.ModuleAccountSource
+		simsx.AccountSource
+	}
+
+	BankKeeper interface {
+		simsx.BalanceSource
+		GetBlockedAddresses() map[string]bool
+	}
+
+	StakingKeeper interface {
+		UnbondingTime(ctx context.Context) (time.Duration, error)
+	}
+
+	ModuleManager interface {
+		Modules() map[string]appmodulev2.AppModule
+	}
+
+	// SimulationApp abstract blockchain app
+	SimulationApp[T Tx] interface {
+		GetApp() *runtime.App[T]
+		ModuleManager() *runtime.MM[T]
```

Comment:
If you have the app you have the module manager. So we actually do not need this.
Unless you expose the appmanager interface directly instead of the runtime.App effectively changing this: https://github.com/cosmos/cosmos-sdk/pull/23013#discussion_r1905775162

Reply:
👍 makes sense!